### ISSUE TITLE
RecipeAdhocSource timing sensitive test has started failing every night

### ISSUE THREAD

**Issue by pjfanning** [apache/incubator-pekko#886]
https://github.com/apache/incubator-pekko/actions/runs/7342976146/job/19992946889

Started 5 days ago. Almost certainly caused by https://github.com/apache/incubator-pekko/pull/861

@He-Pin could you have a look?

```
[12-28 00:41:26.811] [info] - must restart up to specified maxRetries *** FAILED *** (2 seconds, 529 milliseconds)
[12-28 00:41:26.811] [info]   class org.apache.pekko.stream.BackpressureTimeoutException was not equal to class java.util.concurrent.TimeoutException (RecipeAdhocSource.scala:130)
[12-28 00:41:26.812] [info]   org.scalatest.exceptions.TestFailedException:
[12-28 00:41:26.812] [info]   at org.scalatest.matchers.MatchersHelper$.indicateFailure(MatchersHelper.scala:397)
[12-28 00:41:26.812] [info]   at org.scalatest.matchers.should.Matchers$ShouldMethodHelperClass.shouldMatcher(Matchers.scala:7299)
[12-28 00:41:26.812] [info]   at org.scalatest.matchers.should.Matchers$AnyShouldWrapper.should(Matchers.scala:7347)
[12-28 00:41:26.812] [info]   at docs.stream.cookbook.RecipeAdhocSource.$anonfun$new$12(RecipeAdhocSource.scala:130)
[12-28 00:41:26.812] [info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[12-28 00:41:26.812] [info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[12-28 00:41:26.812] [info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[12-28 00:41:26.812] [info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[12-28 00:41:26.813] [info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[12-28 00:41:26.813] [info]   at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1240)
[12-28 00:41:26.813] [info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[12-28 00:41:26.813] [info]   at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[12-28 00:41:26.813] [info]   at org.apache.pekko.testkit.PekkoSpec.withFixture(PekkoSpec.scala:80)
[12-28 00:41:26.813] [info]   at org.scalatest.wordspec.AnyWordSpecLike.invokeWithFixture$1(AnyWordSpecLike.scala:1238)
[12-28 00:41:26.813] [info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTest$1(AnyWordSpecLike.scala:1250)
[12-28 00:41:26.813] [info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[12-28 00:41:26.814] [info]   at org.scalatest.wordspec.AnyWordSpecLike.runTest(AnyWordSpecLike.scala:1250)
[12-28 00:41:26.817] [info]   at org.scalatest.wordspec.AnyWordSpecLike.runTest$(AnyWordSpecLike.scala:1232)
[12-28 00:41:26.820] [info]   at org.apache.pekko.testkit.PekkoSpec.runTest(PekkoSpec.scala:80)
[12-28 00:41:26.820] [info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTests$1(AnyWordSpecLike.scala:1309)
[12-28 00:41:26.820] [info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
[12-28 00:41:26.820] [info]   at scala.collection.immutable.List.foreach(List.scala:333)
[12-28 00:41:26.821] [info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[12-28 00:41:26.821] [info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:390)
[12-28 00:41:26.821] [info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:427)
[12-28 00:41:26.821] [info]   at scala.collection.immutable.List.foreach(List.scala:333)
[12-28 00:41:26.821] [info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[12-28 00:41:26.821] [info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[12-28 00:41:26.821] [info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[12-28 00:41:26.821] [info]   at org.scalatest.wordspec.AnyWordSpecLike.runTests(AnyWordSpecLike.scala:1309)
[12-28 00:41:26.821] [info]   at org.scalatest.wordspec.AnyWordSpecLike.runTests$(AnyWordSpecLike.scala:1308)
[12-28 00:41:26.821] [info]   at org.apache.pekko.testkit.PekkoSpec.runTests(PekkoSpec.scala:80)
[12-28 00:41:26.821] [info]   at org.scalatest.Suite.run(Suite.scala:1114)
[12-28 00:41:26.821] [info]   at org.scalatest.Suite.run$(Suite.scala:1096)
[12-28 00:41:26.821] [info]   at org.apache.pekko.testkit.PekkoSpec.org$scalatest$wordspec$AnyWordSpecLike$$super$run(PekkoSpec.scala:80)
[12-28 00:41:26.821] [info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$run$1(AnyWordSpecLike.scala:1354)
[12-28 00:41:26.821] [info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[12-28 00:41:26.821] [info]   at org.scalatest.wordspec.AnyWordSpecLike.run(AnyWordSpecLike.scala:1354)
[12-28 00:41:26.821] [info]   at org.scalatest.wordspec.AnyWordSpecLike.run$(AnyWordSpecLike.scala:1352)
[12-28 00:41:26.821] [info]   at org.apache.pekko.testkit.PekkoSpec.org$scalatest$BeforeAndAfterAll$$super$run(PekkoSpec.scala:80)
[12-28 00:41:26.821] [info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
[12-28 00:41:26.821] [info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
Coroner Thread Count started at 100, ended at 100, peaked at 104 in docs.stream.cookbook.RecipeAdhocSource
[12-28 00:41:26.865] [info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[12-28 00:41:26.865] [info]   at org.apache.pekko.testkit.PekkoSpec.run(PekkoSpec.scala:80)
[12-28 00:41:26.865] [info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
[12-28 00:41:26.865] [info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
[12-28 00:41:26.865] [info]   at sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:414)
[12-28 00:41:26.865] [info]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[12-28 00:41:26.865] [info]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[12-28 00:41:26.865] [info]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[12-28 00:41:26.865] [info]   at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by He-Pin** [comment]
Ok

**Comment by pjfanning** [comment]
fixed via #888
